File: COM/Samples/Dot Net/VS2005/C#/Sample1/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.IO;

namespace Sample1
{
	public class Form1 : System.Windows.Forms.Form
	{
		private const int maxTime  = 20;

		private PDFCreator.clsPDFCreator _PDFCreator;
		private PDFCreator.clsPDFCreatorError pErr;

        private bool ReadyState;
		private System.Windows.Forms.Button button1;
		private System.Windows.Forms.Button button2;
		private System.Windows.Forms.Button button3;
		private System.Windows.Forms.Button button4;
		private System.Windows.Forms.Button button5;
		private System.Windows.Forms.Button button6;
		private System.Windows.Forms.Timer timer1;
		private System.Windows.Forms.OpenFileDialog openFileDialog1;
        private StatusStrip statusStrip1;
        private ToolStripStatusLabel toolStripStatusLabel1;
		private System.ComponentModel.IContainer components;
	
		public Form1()
		{
			InitializeComponent();
		}
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Vom Windows Form-Designer generierter Code
		private void InitializeComponent()
		{
            this.components = new System.ComponentModel.Container();
            this.button1 = new System.Windows.Forms.Button();
            this.button2 = new System.Windows.Forms.Button();
            this.button3 = new System.Windows.Forms.Button();
            this.button4 = new System.Windows.Forms.Button();
            this.button5 = new System.Windows.Forms.Button();
            this.button6 = new System.Windows.Forms.Button();
            this.timer1 = new System.Windows.Forms.Timer(this.components);
            this.openFileDialog1 = new System.Windows.Forms.OpenFileDialog();
            this.statusStrip1 = new System.Windows.Forms.StatusStrip();
            this.toolStripStatusLabel1 = new System.Windows.Forms.ToolStripStatusLabel();
            this.statusStrip1.SuspendLayout();
            this.SuspendLayout();
            // 
            // button1
            // 
            this.button1.Enabled = false;
            this.button1.Location = new System.Drawing.Point(8, 8);
            this.button1.Name = "button1";
            this.button1.Size = new System.Drawing.Size(152, 40);
            this.button1.TabIndex = 0;
            this.button1.Text = "Show options dialog";
            this.button1.Click += new System.EventHandler(this.button1_Click);
            // 
            // button2
            // 
            this.button2.Enabled = false;
            this.button2.Location = new System.Drawing.Point(8, 56);
            this.button2.Name = "button2";
            this.button2.Size = new System.Drawing.Size(152, 40);
            this.button2.TabIndex = 1;
            this.button2.Text = "Show logfile dialog";
            this.button2.Click += new System.EventHandler(this.button2_Click);
            // 
            // button3
            // 
            this.button3.Enabled = false;
            this.button3.Location = new System.Drawing.Point(192, 8);
            this.button3.Name = "button3";
            this.button3.Size = new System.Drawing.Size(152, 40);
            this.button3.TabIndex = 2;
            this.button3.Text = "Print printer testpage";
            this.button3.Click += new System.EventHandler(this.button3_Click);
            // 
            // button4
            // 
            this.button4.Enabled = false;
            this.button4.Location = new System.Drawing.Point(192, 56);
            this.button4.Name = "button4";
            this.button4.Size = new System.Drawing.Size(152, 40);
            this.button4.TabIndex = 3;
            this.button4.Text = "Print PDFCreator testpage";
            this.button4.Click += new System.EventHandler(this.button4_Click);
            // 
            // button5
            // 
            this.button5.Enabled = false;
            this.button5.Location = new System.Drawing.Point(376, 8);
            this.button5.Name = "button5";
            this.button5.Size = new System.Drawing.Size(152, 40);
            this.button5.TabIndex = 4;
            this.button5.Text = "Convert to PDF";
            this.button5.Click += new System.EventHandler(this.button5_Click);
            // 
            // button6
            // 
            this.button6.Enabled = false;
            this.button6.Location = new System.Drawing.Point(376, 56);
            this.button6.Name = "button6";
            this.button6.Size = new System.Drawing.Size(152, 40);
            this.button6.TabIndex = 5;
            this.button6.Text = "Convert to TIFF";
            this.button6.Click += new System.EventHandler(this.button6_Click);
            // 
            // timer1
            // 
            this.timer1.Tick += new System.EventHandler(this.timer1_Tick);
            // 
            // statusStrip1
            // 
            this.statusStrip1.Items.AddRange(new System.Windows.Forms.ToolStripItem[] {
            this.toolStripStatusLabel1});
            this.statusStrip1.Location = new System.Drawing.Point(0, 112);
            this.statusStrip1.Name = "statusStrip1";
            this.statusStrip1.Size = new System.Drawing.Size(536, 22);
            this.statusStrip1.TabIndex = 6;
            this.statusStrip1.Text = "statusStrip1";
            // 
            // toolStripStatusLabel1
            // 
            this.toolStripStatusLabel1.Name = "toolStripStatusLabel1";
            this.toolStripStatusLabel1.Size = new System.Drawing.Size(42, 17);
            this.toolStripStatusLabel1.Text = "Status:";
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(536, 134);
            this.Controls.Add(this.statusStrip1);
            this.Controls.Add(this.button1);
            this.Controls.Add(this.button2);
            this.Controls.Add(this.button3);
            this.Controls.Add(this.button4);
            this.Controls.Add(this.button5);
            this.Controls.Add(this.button6);
            this.Name = "Form1";
            this.Text = "Sample1 - PDFCreator COM interface";
            this.Closing += new System.ComponentModel.CancelEventHandler(this.Form1_Closing);
            this.Load += new System.EventHandler(this.Form1_Load);
            this.statusStrip1.ResumeLayout(false);
            this.statusStrip1.PerformLayout();
            this.ResumeLayout(false);
            this.PerformLayout();

		}
		#endregion

		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		private void Form1_Load(object sender, System.EventArgs e)
		{
			string parameters;
            toolStripStatusLabel1.Text = "Status: Program is started.";

			pErr = new PDFCreator.clsPDFCreatorError();

			_PDFCreator = new PDFCreator.clsPDFCreator();
			_PDFCreator.eError += new PDFCreator.__clsPDFCreator_eErrorEventHandler(_PDFCreator_eError); 
			_PDFCreator.eReady += new PDFCreator.__clsPDFCreator_eReadyEventHandler(_PDFCreator_eReady); 
        
			parameters = "/NoProcessingAtStartup";

			if (!_PDFCreator.cStart(parameters, false))
			{
                toolStripStatusLabel1.Text = "Status: Error[" + pErr.Number + "]: " + pErr.Description;
            }
			else
			{
				button1.Enabled = true;
				button2.Enabled = true;
				button3.Enabled = true;
				button4.Enabled = true;
				button5.Enabled = true;
				button6.Enabled = true;
			}
		}

		private void button1_Click(object sender, System.EventArgs e)
		{
            _PDFCreator.cShowOptionsDialog(true);
		}

		private void button2_Click(object sender, System.EventArgs e)
		{
			_PDFCreator.cShowLogfileDialog(true);
		}

		private void button3_Click(object sender, System.EventArgs e)
		{
			_PDFCreator.cDefaultPrinter = "PDFCreator";
			// Wait 1 second
			timer1.Interval = 1000;
			timer1.Enabled = true;
			while (timer1.Enabled)
			{
				Application.DoEvents();
			}
			_PDFCreator.cPrinterStop = false;
			_PDFCreator.cPrintPrinterTestpage("");
		}

		private void button4_Click(object sender, System.EventArgs e)
		{
			_PDFCreator.cPrintPDFCreatorTestpage();
			_PDFCreator.cPrinterStop = true;
		}

		private void button5_Click(object sender, System.EventArgs e)
		{
	        PrintIt(0);
		}

		private void button6_Click(object sender, System.EventArgs e)
		{
	        PrintIt(0);
		}

		private void PrintIt(int FileTyp)
		{
			string fname, DefaultPrinter;
			FileInfo fi;
			PDFCreator.clsPDFCreatorOptions opt;
			openFileDialog1.Multiselect = false;
			openFileDialog1.CheckFileExists = true;
			openFileDialog1.CheckPathExists = true;
			openFileDialog1.Filter = "Text files (*.txt)|*.txt|All files (*.*)|*.*";
			if (openFileDialog1.ShowDialog() == DialogResult.OK)
			{
				fi = new FileInfo(openFileDialog1.FileName);
				if (fi.Name.Length > 0)
				{
					if (fi.Name.IndexOf(".") > 1)
					{
						fname = fi.Name.Substring(0,fi.Name.IndexOf("."));
					}
					else
					{
						fname = fi.Name;
					}
					if (!_PDFCreator.cIsPrintable(fi.FullName))
					{
						MessageBox.Show("File '" + fi.FullName + "' is not printable!", this.Text,MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
						return;
					}
					opt = _PDFCreator.cOptions;
					opt.UseAutosave = 1;
					opt.UseAutosaveDirectory = 1;
					opt.AutosaveDirectory = fi.DirectoryName;
					opt.AutosaveFormat = FileTyp;
					if (FileTyp == 5)
					{
						opt.BitmapResolution = 72;
					}
					opt.AutosaveFilename = fname;
					_PDFCreator.cOptions = opt;
					_PDFCreator.cClearCache();
					DefaultPrinter = _PDFCreator.cDefaultPrinter;
					_PDFCreator.cDefaultPrinter = "PDFCreator";
					_PDFCreator.cPrintFile(fi.FullName);
                    ReadyState = false;
                    _PDFCreator.cPrinterStop = false;
					timer1.Interval = maxTime * 1000;
					timer1.Enabled = true;
					while (!ReadyState && timer1.Enabled)
					{
						Application.DoEvents();
					}
					timer1.Enabled = false;
					if (!ReadyState)
					{
						MessageBox.Show("Creating printer test page as pdf.\n\r\n\r" + 
							"An error is occured: Time is up!", this.Text,MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
					}
					_PDFCreator.cPrinterStop = true;
					_PDFCreator.cDefaultPrinter = DefaultPrinter;
				}
			}
		}

		private void _PDFCreator_eReady()
		{
            toolStripStatusLabel1.Text = "Status: \"" + _PDFCreator.cOutputFilename + "\" was created!"; 
            _PDFCreator.cPrinterStop = true;
			ReadyState = true;
		}

		private void _PDFCreator_eError()
		{
			pErr = _PDFCreator.cError;
		}

		private void timer1_Tick(object sender, System.EventArgs e)
		{
			timer1.Enabled=false;
		}

		private void Form1_Closing(object sender, System.ComponentModel.CancelEventArgs e)
		{
			_PDFCreator.cClose();
			System.Runtime.InteropServices.Marshal.ReleaseComObject(_PDFCreator);
			System.Runtime.InteropServices.Marshal.ReleaseComObject(pErr);
			pErr = null;
			GC.Collect();
            GC.WaitForPendingFinalizers();
            GC.Collect();
            GC.WaitForPendingFinalizers();
        }
	}
}

File: tags/Version 0.9.6/COM/Samples/Dot Net/VS2003/Visual Basic/Sample1/Form1.vb
Imports System.IO

Public Class Form1
    Inherits System.Windows.Forms.Form

    Private Const maxTime As Long = 20

    Private WithEvents _PDFCreator As PDFCreator.clsPDFCreator
    Private pErr As PDFCreator.clsPDFCreatorError

    Private ReadyState As Boolean

#Region " Vom Windows Form Designer generierter Code "

    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    Private components As System.ComponentModel.IContainer

    Friend WithEvents StatusBar1 As System.Windows.Forms.StatusBar
    Friend WithEvents Button1 As System.Windows.Forms.Button
    Friend WithEvents Button2 As System.Windows.Forms.Button
    Friend WithEvents Button3 As System.Windows.Forms.Button
    Friend WithEvents Button4 As System.Windows.Forms.Button
    Friend WithEvents Button5 As System.Windows.Forms.Button
    Friend WithEvents Button6 As System.Windows.Forms.Button
    Friend WithEvents Timer1 As System.Windows.Forms.Timer
    Friend WithEvents OpenFileDialog1 As System.Windows.Forms.OpenFileDialog
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container
        Me.StatusBar1 = New System.Windows.Forms.StatusBar
        Me.Button1 = New System.Windows.Forms.Button
        Me.Button2 = New System.Windows.Forms.Button
        Me.Button3 = New System.Windows.Forms.Button
        Me.Button4 = New System.Windows.Forms.Button
        Me.Button5 = New System.Windows.Forms.Button
        Me.Button6 = New System.Windows.Forms.Button
        Me.Timer1 = New System.Windows.Forms.Timer(Me.components)
        Me.OpenFileDialog1 = New System.Windows.Forms.OpenFileDialog
        Me.SuspendLayout()
        '
        'StatusBar1
        '
        Me.StatusBar1.Location = New System.Drawing.Point(0, 118)
        Me.StatusBar1.Name = "StatusBar1"
        Me.StatusBar1.Size = New System.Drawing.Size(536, 16)
        Me.StatusBar1.TabIndex = 2
        Me.StatusBar1.Text = "Status:"
        '
        'Button1
        '
        Me.Button1.Enabled = False
        Me.Button1.Location = New System.Drawing.Point(8, 8)
        Me.Button1.Name = "Button1"
        Me.Button1.Size = New System.Drawing.Size(152, 40)
        Me.Button1.TabIndex = 0
        Me.Button1.Text = "Show options dialog"
        '
        'Button2
        '
        Me.Button2.Enabled = False
        Me.Button2.Location = New System.Drawing.Point(8, 56)
        Me.Button2.Name = "Button2"
        Me.Button2.Size = New System.Drawing.Size(152, 40)
        Me.Button2.TabIndex = 1
        Me.Button2.Text = "Show logfile dialog"
        '
        'Button3
        '
        Me.Button3.Enabled = False
        Me.Button3.Location = New System.Drawing.Point(192, 8)
        Me.Button3.Name = "Button3"
        Me.Button3.Size = New System.Drawing.Size(152, 40)
        Me.Button3.TabIndex = 3
        Me.Button3.Text = "Print printer testpage"
        '
        'Button4
        '
        Me.Button4.Enabled = False
        Me.Button4.Location = New System.Drawing.Point(192, 56)
        Me.Button4.Name = "Button4"
        Me.Button4.Size = New System.Drawing.Size(152, 40)
        Me.Button4.TabIndex = 4
        Me.Button4.Text = "Print PDFCreator testpage"
        '
        'Button5
        '
        Me.Button5.Enabled = False
        Me.Button5.Location = New System.Drawing.Point(376, 8)
        Me.Button5.Name = "Button5"
        Me.Button5.Size = New System.Drawing.Size(152, 40)
        Me.Button5.TabIndex = 5
        Me.Button5.Text = "Convert to PDF"
        '
        'Button6
        '
        Me.Button6.Enabled = False
        Me.Button6.Location = New System.Drawing.Point(376, 56)
        Me.Button6.Name = "Button6"
        Me.Button6.Size = New System.Drawing.Size(152, 40)
        Me.Button6.TabIndex = 6
        Me.Button6.Text = "Convert to TIFF"
        '
        'Timer1
        '
        '
        'Form1
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(536, 134)
        Me.Controls.Add(Me.Button6)
        Me.Controls.Add(Me.Button5)
        Me.Controls.Add(Me.Button4)
        Me.Controls.Add(Me.Button3)
        Me.Controls.Add(Me.Button2)
        Me.Controls.Add(Me.Button1)
        Me.Controls.Add(Me.StatusBar1)
        Me.Name = "Form1"
        Me.Text = "Sample1 - PDFCreator COM interface"
        Me.ResumeLayout(False)

    End Sub

#End Region

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim parameters As String
        StatusBar1.Text = "Status: Program is started."

        pErr = New PDFCreator.clsPDFCreatorError
        _PDFCreator = New PDFCreator.clsPDFCreator

        parameters = "/NoProcessingAtStartup"

        If _PDFCreator.cStart(parameters) = False Then
            StatusBar1.Text = "Status: Error[" & pErr.Number & "]: " & pErr.Description
        Else
            Button1.Enabled = True
            Button2.Enabled = True
            Button3.Enabled = True
            Button4.Enabled = True
            Button5.Enabled = True
            Button6.Enabled = True
        End If
    End Sub

    Private Sub PDFCreator_Ready() Handles _PDFCreator.eReady
        StatusBar1.Text = "Status: """ & _PDFCreator.cOutputFilename & """ was created!"
        _PDFCreator.cPrinterStop = True
        ReadyState = True
    End Sub

    Private Sub _PDFCreator_eError() Handles _PDFCreator.eError
        pErr = _PDFCreator.cError
    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        Timer1.Enabled = False
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        _PDFCreator.cShowOptionsDialog(True)
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        _PDFCreator.cShowLogfileDialog(True)
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        _PDFCreator.cDefaultPrinter = "PDFCreator"
        ' Wait 1 second
        Timer1.Interval = 1000
        Timer1.Enabled = True
        Do While Timer1.Enabled
            Application.DoEvents()
        Loop
        _PDFCreator.cPrinterStop = False
        _PDFCreator.cPrintPrinterTestpage()
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        _PDFCreator.cPrintPDFCreatorTestpage()
        _PDFCreator.cPrinterStop = False
    End Sub

    Private Sub Button5_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button5.Click
        PrintIt(0)
    End Sub

    Private Sub Button6_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button6.Click
        PrintIt(5)
    End Sub

    Private Sub PrintIt(ByVal Filetyp As Long)
        Dim fname As String, fi As FileInfo, DefaultPrinter As String
        Dim opt As PDFCreator.clsPDFCreatorOptions
        With OpenFileDialog1
            .Multiselect = False
            .CheckFileExists = True
            .CheckPathExists = True
            .Filter = "Text files (*.txt)|*.txt|All files (*.*)|*.*"
        End With
        If OpenFileDialog1.ShowDialog = DialogResult.OK Then
            fi = New FileInfo(OpenFileDialog1.FileName)
            If fi.Name.Length > 0 Then
                If InStr(fi.Name, ".", CompareMethod.Text) > 1 Then
                    fname = Mid(fi.Name, 1, InStr(fi.Name, ".", CompareMethod.Text) - 1)
                Else
                    fname = fi.Name
                End If
            End If
            If Not _PDFCreator.cIsPrintable(fi.FullName) Then
                MsgBox("File '" & fi.FullName & "' is not printable!", MsgBoxStyle.Exclamation, Me.Text)
                Exit Sub
            End If
            opt = _PDFCreator.cOptions
            With opt
                .UseAutosave = 1
                .UseAutosaveDirectory = 1
                .AutosaveDirectory = fi.DirectoryName
                .AutosaveFormat = Filetyp
                If Filetyp = 5 Then
                    .BitmapResolution = 72
                End If
                opt.AutosaveFilename = fname
            End With
            With _PDFCreator
                .cOptions = opt
                .cClearCache()
                DefaultPrinter = .cDefaultPrinter
                .cDefaultPrinter = "PDFCreator"
                .cPrintFile(fi.FullName)
                ReadyState = False
                .cPrinterStop = False
            End With

            With Timer1
                .Interval = maxTime * 1000
                .Enabled = True
                Do While Not ReadyState And .Enabled
                    Application.DoEvents()
                Loop
                .Enabled = False
            End With
            If Not ReadyState Then
                MsgBox("Creating printer test page as pdf." & vbCrLf & vbCrLf & _
                 "An error is occured: Time is up!", MsgBoxStyle.Exclamation, Me.Text)
            End If
            _PDFCreator.cPrinterStop = True
            _PDFCreator.cDefaultPrinter = DefaultPrinter
        End If
        opt = Nothing
    End Sub

    Private Sub Form1_Closing(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles MyBase.Closing
        _PDFCreator.cClose()
        System.Runtime.InteropServices.Marshal.ReleaseComObject(_PDFCreator)
        System.Runtime.InteropServices.Marshal.ReleaseComObject(pErr)
        pErr = Nothing
        _PDFCreator = Nothing
        GC.Collect()
        GC.WaitForPendingFinalizers()
    End Sub
End Class